src/Raven.Server/Documents/Queries/Results/TimeSeries/TimeSeriesRetriever.cs

```diff
@@ -1275,7 +1275,11 @@ public static DateTime ParseDateTime(string valueAsStr)
                 throw new ArgumentException($"Unable to parse timeseries from/to values. Got: {valueAsStr}{Environment.NewLine}" +
                                             $"The supported time formats are:{Environment.NewLine}" +
                                             $"{string.Join(Environment.NewLine, SupportedDateTimeFormats.OrderBy(f => f.Length))}");
-            return DateTime.SpecifyKind(date, DateTimeKind.Utc);
+
+            if (date == DateTime.MaxValue || date == DateTime.MinValue)
+                return DateTime.SpecifyKind(date, DateTimeKind.Utc);
+
+            return new DateTime(date.Year, date.Month, date.Day, date.Hour, date.Minute, date.Second, date.Millisecond, DateTimeKind.Utc);
```

Comment:
I think we should use `EnsureMillisecondsPrecision` instead

Reply:
way better, done :-)